src/index.md

```diff
@@ -12,6 +12,118 @@ The goal of this guide is to answer questions that developers of the JDK might h
 
 There are many common use cases that aren't detailed in the formal process. This guide suggests how to work in such cases.
 
+## I have a patch, what do I do?
+
+::: {.box}
+[Quick Links]{.boxheader}
+
+* [Oracle Contributor Agreement (OCA)](https://www.oracle.com/technical-resources/oracle-contributor-agreement.html)
+* [JDK Bug System (JBS)](https://bugs.openjdk.java.net/)
+* [OpenJDK Project Roles](https://openjdk.java.net/bylaws#project-roles)
+:::
+
+In many GitHub projects the standard way to propose a change is to create a pull request (PR) and discuss the patch in the PR. For OpenJDK projects the situation is somewhat different. First of all, the JDK and the surrounding tooling developed in the OpenJDK projects are part of commercial products maintained and sold by many companies. The JDK is installed on billions of devices and there are many millions of developers out there who use Java. For this reason there must be several safety checks before a change is made.
+
+### 1. Sign the OCA
+
+Oracle is the steward of the OpenJDK project, and owns the Java brand. In order to make your patch available for review you must first sign the [Oracle Contributor Agreement](https://www.oracle.com/technical-resources/oracle-contributor-agreement.html) (OCA). This agreement gives Oracle and you as a contributor joint copyright interests in the code. You will retain your copyright while also granting those rights to Oracle.
+
+When you sign the OCA, please make sure that you specify your GitHub user name in the `Username` field of the OCA. If you try to create a PR before you have signed the OCA, or if you didn't specify your GitHub user name, you'll get instructions telling you to do so, and the PR won't be published until this is done. OCA registration is a manual process. Please allow for up to several days to have your OCA application processed, even though it's normally processed swiftly.
+
+### 2. Socialize your change
+
+Once the OCA is signed, please restrain your urge to create a PR just a little while longer. In order to prepare the community for your patch, please socialize your idea on the relevant [mailing lists](#mailing-lists). Almost all changes, and in particular any API changes, must go this route and have a broad agreement in place before there is any point in presenting code. To understand the criteria by which your patch is going to be judged, please read [_Why is My Change Rejected?_](#why-is-my-change-rejected) below. In short, hidden constraints and assumptions, stability and quality, maintainability, compatibility, and conformance to specifications must be considered before your PR is ready to be submitted. If you don't understand the constraints for acceptance, you might be surprised when your PR is rejected.
```

Comment:
Thanks for the update. One small suggestion is to say "proposals for new APIs" rather than "any API change".

Reply:
That seems like it narrows down the criteria. Maybe Random J Hacker only wants to add a getFronglified() to java.lang.String -- since that is not a new API, he does not think it necessary to discuss it first..?

I'd say keep it as it is; it's better to send one too many to the mailing list rather than one too few.

And also, it's better to actually get this document out to the public. We can refine it iteratively if we discover that people misunderstand it.

Reply:
String::getFronglified would be a new API, best to get agreement on the mailing list before proposing a patch. It's very possible that the API has been discussed and rejected many times or it has interactions with many other areas of the platform. To some extent we have a bit of a cultural clash here in that many developers are used to "code first" and use the PR to start the discussion. For new standard or JDK-specific APIs, this might be the first 1% of the work.

Reply:
I tend to agree with Magnus here. Don't we want to cover changes that removes a part of the API och just changes the behavior of some method?

Reply:
I think "any API change" is preferable to "new API" -- I could easily imagine somebody coming along and saying, "All I want to do is to have this method return null instead of throwing NPE. That's not a new API." There's a larger discussion to be had somewhere else about what constitutes an API change, in particular that some behavior changes *are* API changes. Meanwhile "any API change" is probably sufficiently inclusive for now.

Reply:
Okay but Magnus's example is a new API.